vmss.tf

```diff
@@ -73,7 +74,100 @@ resource "azurerm_proximity_placement_group" "ppg" {
   tags                = merge(var.tags_map, {"weka_cluster": var.cluster_name})
 }
 
-resource "azurerm_linux_virtual_machine_scale_set" "vmss" {
+data "azurerm_shared_image_version" "shared_image_version" {
+ count               = length(var.custom_image) > 0 ? 1 : 0
+ gallery_name        = lookup(var.custom_image, "gallery_name", null)
+ image_name          = lookup(var.custom_image, "definition_name", null)
+ name                = lookup(var.custom_image, "version", null)
+ resource_group_name = lookup(var.custom_image, "rg_name", null)
+}
+
+data azurerm_platform_image "image" {
```

Comment:
you don't really need it, you anyway do lookup, so why not just do it on the vmss instead of having this data.

Reply:
right, but if AZURE will give me answer maybe we will need it for custom image and default image, but for now you are right!